lib/matplotlib/ticker.py

```diff
@@ -1434,7 +1434,7 @@ def set_params(self, **kwargs):
     def bin_boundaries(self, vmin, vmax):
         nbins = self._nbins
         if nbins == 'auto':
-            nbins = min(self.axis.get_tick_space(), 9)
+            nbins = max(min(self.axis.get_tick_space(), 9), 1)
```

Comment:
Sounds like a reasonable guess, but I think  that at least originally, nbins was a *maximum* number of *intervals*, so the maximum number of ticks would be one larger; but all of this is dealing with maxima, not with any guaranteed number.